.buildkite/run-amd-test.sh

```diff
@@ -70,16 +71,51 @@ HF_CACHE="$(realpath ~)/huggingface"
 mkdir -p ${HF_CACHE}
 HF_MOUNT="/root/.cache/huggingface"
 
-docker run \
+commands=$@
+PARALLEL_JOB_COUNT=8
+#check if the command contains shard flag
+if [[ $commands == *"--shard-id="* ]]; then
+  for GPU in $(seq 0 $(($PARALLEL_JOB_COUNT-1))); do
```

Comment:
This is an incorrect implementation of the sharding. Buildkite should already started X number of jobs under the same name. Each run script should just receive the environment variable, and pass it along to the command. 

The current implementation is trying to run all shards in the same command

Reply:
in more detail, it looks like the they are indeed running in the same job in parallel 
https://buildkite.com/vllm/ci-aws/builds/7784#01919a58-e1eb-48b3-9fd5-872f0328e913

this might break more often than we wanted?

Reply:
This invocation is compatible with the *general* way we launch tests. IMHO unless there is  a problem with execution of the "payload" tests, we shouldn't be restricted in the way we implement the invocation logic.

Reply:
Our engineering choices are dictated by the specific nature of our HW infrastructure and its initialization/decoupling.

Reply:
We cannot use docker Buildkite plugin, so we have parallelize the jobs ourselves. Our shell script receives the command with empty "--shard-id=" argument, so we have to substitute it and run as background jobs while exposing one GPU to each job.

Reply:
sorry how many GPUs is there on each node? You can run multiple buildkite agent on the host and pin each to a GPU using environment variable. This can drastically help accelerate the test

Reply:
my main concern with this approach is now the way sharding is handling is implemented differently and can cause issues when developers are debugging the test failures on amd devices.

Reply:
> We cannot use docker Buildkite plugin
Can I ask what's the reason? For for sharding, it is not necessary to use the plugin. Sharding is a native option. see https://buildkite.com/docs/tutorials/parallel-builds

Reply:
Each node has 8 GPUs. The restarting procedure is indiscriminate though,- we're restarting *all* GPUs on a given node at once. This strategy has advantage of complete between-test decoupling. The unfortunate downside is that we can't rely on multiple Buildkite agents running on the same host.

We achieved the current level of HW stability with this approach.

Reply:
Got it. Let's refine this PR a bit and we can merge it in